internal/pkg/manifest/lb_web_svc.go

```diff
@@ -112,23 +100,26 @@ func (s *LoadBalancedWebService) MarshalBinary() ([]byte, error) {
 
 // DockerfilePath returns the image build path.
 func (s *LoadBalancedWebService) DockerfilePath() string {
-	return s.Image.Build
+	return aws.StringValue(s.Image.Build)
 }
 
 // ApplyEnv returns the service manifest with environment overrides.
 // If the environment passed in does not have any overrides then it returns itself.
-func (s *LoadBalancedWebService) ApplyEnv(envName string) *LoadBalancedWebService {
-	target, ok := s.Environments[envName]
+func (s LoadBalancedWebService) ApplyEnv(envName string) (*LoadBalancedWebService, error) {
+	overrideConfig, ok := s.Environments[envName]
 	if !ok {
-		return s
+		return &s, nil
 	}
-
-	return &LoadBalancedWebService{
-		Service:     s.Service,
-		Image:       target.Image,
-		RoutingRule: s.RoutingRule.copyAndApply(target.RoutingRule),
-		TaskConfig:  s.TaskConfig.copyAndApply(target.TaskConfig),
-		Sidecar:     s.Sidecar.copyAndApply(target.Sidecar),
-		LogConfig:   s.LogConfig.copyAndApply(target.LogConfig),
+	target := LoadBalancedWebService{
+		Image:       overrideConfig.Image,
+		RoutingRule: overrideConfig.RoutingRule,
+		TaskConfig:  overrideConfig.TaskConfig,
+		LogConfig:   overrideConfig.LogConfig,
+		Sidecar:     overrideConfig.Sidecar,
+	}
+	if err := mergo.Merge(&s, target, mergo.WithOverride, mergo.WithOverwriteWithEmptyValue); err != nil {
```

Comment:
[`mergo.WithOverwriteWithEmptyValue`](https://github.com/imdario/mergo/blob/9080aaff85364c8348d7086f09cca7f803e14822/merge.go#L310-L314) is a superset of `mergo.WithOverride`. We can remove the latter

Reply:
it's not :| I tried to remove that one and the unit test failed because it was not overriden